app/controllers/user_challenges_controller.rb

```diff
@@ -38,6 +39,6 @@ def set_user
   end
 
   def set_user_challenge
-    @user_challenge = UserChallenge.find(params[:id])
+    @user_challenge = current_user.user_challenges.find(params[:id])
```

Comment:
I guess this is your way of not having to have a user for each note?

Reply:
It prevents from a user seeing other user's user_challenges by any accident!